SYCL/ESIMD/api/functional/ctors/ctor_load_core.cpp

```diff
@@ -0,0 +1,256 @@
+    // one value from reference data.
+    // If current number of elements is greater than one, then run tests with
+    // whole reference data.
+    if constexpr (NumElems == 1) {
+      for (size_t i = 0; i < ref_data.size(); ++i) {
+        passed = run_verification(queue, {ref_data[i]}, data_type);
+      }
+    } else {
+      passed = run_verification(queue, ref_data, data_type);
+    }
+    return passed;
+  }
+
+private:
+  bool run_verification(sycl::queue &queue, const std::vector<DataT> &ref_data,
+                        const std::string &data_type) {
+    assert(ref_data.size() == NumElems &&
+           "Reference data size is not equal to the simd vector length.");
+
+    bool passed = true;
+
+    const std::size_t alignment_value =
+        get_alignment_value<DataT, NumElems, AlignmentT::value>();
+    const std::size_t container_extra_size =
+        alignment_value / sizeof(DataT) + 1;
+
+    shared_allocator<DataT> allocator(queue);
+    shared_vector<DataT> result(NumElems, allocator);
+    shared_vector<DataT> shared_ref_data(NumElems + container_extra_size,
+                                         shared_allocator<DataT>(queue));
+
+    size_t object_size = NumElems * sizeof(DataT);
+    size_t buffer_size = object_size + alignment_value;
+
+    // Using pointer to next element to avoid bigger alignment of USM shared
+    // memory to guarantee the weakest alignment. By thi std::align can provide
+    // expected alignment on the small values of an alignment.
+    void *ref = shared_ref_data.data() + 1;
+    if (std::align(alignment_value, object_size, ref, buffer_size) == nullptr) {
+      return false;
+    }
+    DataT *const ref_aligned = static_cast<DataT *>(ref);
+
+    for (size_t i = 0; i < NumElems; ++i) {
+      ref_aligned[i] = ref_data[i];
+    }
+
+    auto alignment = get_alignment<DataT, NumElems, AlignmentT::value>();
+    queue.submit([&](sycl::handler &cgh) {
+      DataT *const out = result.data();
+
+      cgh.single_task<Kernel1<DataT, NumElems, TestCaseT, AlignmentT>>(
+          [=]() SYCL_ESIMD_KERNEL {
+            TestCaseT::template call_simd_ctor<DataT, NumElems>(ref_aligned,
+                                                                out, alignment);
+          });
+    });
+    queue.wait_and_throw();
+
+    for (size_t i = 0; i < result.size(); ++i) {
+      if (!are_bitwise_equal(ref_data[i], result[i])) {
+        passed = false;
+
+        const auto description =
+            ctors::TestDescription<DataT, NumElems, TestCaseT>(
+                i, result[i], ref_data[i], data_type);
+        log::fail(description);
+      }
+    }
+
+    return passed;
+  }
+};
+
+int main(int, char **) {
+  sycl::queue queue(esimd_test::ESIMDSelector{},
+                    esimd_test::createExceptionHandler());
+
+  bool passed = true;
+  const auto types = get_tested_types<tested_types::all>();
+  const auto dims = values_pack<1, 16>();
```

Comment:
Any reason not to cover all dimensions here?

Reply:
I assume that 1 and 16 simd dimensions is enough to cover all necessary cases
Now all dimensions will be covered, I've done it [here](https://github.com/intel/llvm-test-suite/pull/769/commits/9de67432a3606acda7db15caab8cc4c7ef03c3f4#diff-3c39b599d2c35b37e91d3c2ef8b547db1db978189ce11ba0b9ca6e3fff47b5c7R221)